psga/s_aureus/bactopia_one.nf

```diff
@@ -91,8 +96,184 @@ with open('hello.txt', 'w') as outfile:
+        csv_list.append(CsvItem('Strain heterogeneity', checkm_results['Strain heterogeneity']))
+
+    with open('local-single-sample/{sample_id}/assembly/{sample_id}.json'.format(sample_id=sample_id)) as assembly_results_file:
+        assembly_results = json.loads(assembly_results_file.read())
+        for key, value in assembly_results.items():
+            if key == 'fasta':
+                continue
+            csv_list.append(CsvItem(key , value))
+
+    with open('local-single-sample/{sample_id}/antimicrobial-resistance/{sample_id}-protein-report.txt'.format(sample_id=sample_id)) as antimicrobial_protein_file:
+        resistance_genes = list()
+        for row in csv.DictReader(antimicrobial_protein_file, delimiter="\t"):
+            resistance_genes.append(row['Gene symbol'])
+        csv_list.append(CsvItem('Antimicrobial resistance genes', ' '.join(resistance_genes)))
+
+    return csv_list
+
+
+
+def generate_results(
+    output_path: str,
+) -> None:
+    # Get sample_id and genome_assembly_name
+    with open('genome_assembly_name.txt') as genome_assembly_name_file:
+        gaf = json.load(genome_assembly_name_file)
+        genome_assembly_name = gaf['genome_assembly_name']
+        sample_id = gaf['sample_id']
+    print('genome_assembly_name: ' + genome_assembly_name)
+    print('sample_id: ' + sample_id)
+
+    csv_list = build_csv_list(sample_id, genome_assembly_name)
+
+    with open('{sample_id}-results.csv'.format(sample_id=sample_id), 'w', encoding='UTF8') as csv_out:
+        writer = csv.writer(csv_out)
+        writer.writerow([x.header for x in csv_list])
+        writer.writerow([x.value for x in csv_list])
+
+    # Build resultfiles.json
+    json_out = dict()
+    out_files = list()
+    for of in OUTPUT_FILES:
+        # Make sure file and type not already in list (psga framework required no duplicates)
+        # TODO in production version error rather than fixing the problem with a warning
+        file_in_list = False
+        f = of.filepath.format(
+            output_path=output_path, sample_id=sample_id, genome_assembly_name=genome_assembly_name)
+        t = of.typedescription
+        for item in out_files:
+            if item["file"] == f or item["type"] == t:
+                print('file: ', item["file"])
+                print('type: ', item["type"])
+                print('f: ', f)
+                print('t: ', t)
+                file_in_list = True
+        if not file_in_list:
+            out_files.append({
+                "file": f,
+                "type": t
+            })
+        else:
+            print('WARNING: found duplicate file {t} : {f}'.format(t=t, f=f))
+    json_out[sample_id] = out_files
+
+    with open('{sample_id}-resultfiles.json'.format(sample_id=sample_id), 'w') as f:
+        json.dump(json_out, f)
+
+
+if __name__ == "__main__":
+    # pylint: disable=no-value-for-parameter
+    generate_results("${params.output_path}")
```

Comment:
So, if I understand this, you moved the whole code in `generate_results.py` to this nextflow process block. Quite brutal solution, as you could simply have copied this script to the docker image instead enabling one to write unit tests, but at least the bug is gone.

Could you delete the file: `generate_results.py` so that this code is not replicated, please?